### ISSUE TITLE
Searching with a plus sign (+) does not work

### ISSUE THREAD

**Issue by mtwalsh** [aldeed/meteor-tabular#317]
Hi,

Because of the way the searching uses a regex if you try and search with a plus sign (+) it fails.  

A workaround is to escape the plus sign with a backslash but that's not ideal.

**Comment by lynchem** [comment]
Yeah, there are so many characters here that mess up the underlying regex . We should do an escapeRegex on the string passed to mongo. We basically stopped using the inbuilt search mostly but it's worth fixing so you don't see mongo server errors pop up when somebody types `*` into the search box.
